pkg/scheduler/partition.go

```diff
@@ -159,17 +159,19 @@ func (pc *PartitionContext) updatePreemption(conf configs.PartitionConfig) {
 }
 
 func (pc *PartitionContext) updatePartitionDetails(conf configs.PartitionConfig) error {
-	pc.Lock()
-	defer pc.Unlock()
+	// this must be performed without locking to avoid lock order differences between PartitionContext and AppPlacementManager
```

Comment:
"this must be performed without lock" is not 100% correct.  We "**postpone**" the partition locks to avoid lock order differences between PartitionContext and AppPlacementManager.

Reply:
That comments refer to the piece of code which are executed outside the critical section. I'll try to make it more clear.

Reply:
Updated comment.